fix: datetime tryparse (#265)

Arkoniak opened the pull request:
I've tested it locally with Julia versions 1.7.2 and 1.8.1 and it works correctly. Not sure, whether it make sense to add any specific  tests, since `runtests` works with postgres database as a whole and should cover this function (for latest versions of Julia at least).

iamed2 commented:
Yup no need to add specific tests.

This should be added to the ZonedDateTime and UTCDateTime parsing methods as well though, to replicate the same behaviour (I assume it also happens for those types?). For ZonedDateTime you can put the whole for-loop in the try-catch.

Arkoniak commented on src/parsing.jl:
```diff
@@ -259,10 +259,23 @@ function pqparse(::Type{DateTime}, str::AbstractString)
-    isnothing(parsed) || return parsed
+    # Please, do not remove @static, see https://github.com/iamed2/LibPQ.jl/issues/265
+    # for more details
+    @static if v"1.6.6" <= VERSION < v"1.7.0" || VERSION > v"1.7.2"
```
I took this condition from https://github.com/iamed2/LibPQ.jl/blob/master/src/parsing.jl#L318
It seems that the bug only appears in 1.6.6+ and 1.7.1, 1.7.2. This condition is for the proper behavior, i.e. all non bugged versions of Julia.

Arkoniak replied:
Yes, I confirm, I've tested it with 1.8.1 and it is function correctly.

Arkoniak commented:
Well, in the end, I moved all `@static` to utility function, this allows for 
1. having clear code in main function 
2. it'll be easy to remove this fix in the future (just delete `_tryparse` and change all `_tryparse` calls to `tryparse`

Also, I've updated `pqparse(::Type{Time}, str::AbstractString)` function to use the same approach.

I am using `tryparse` in `catch` block, because for some reason for `ZonedDateTime` `tryparse` is working properly in `1.7`

Arkoniak commented:
Well, in the end, I moved all `@static` to utility function, this allows for 
1. having clear code in main function 
2. it'll be easy to remove this fix in the future (just delete `_tryparse` and change all `_tryparse` calls to `tryparse`

Also, I've updated `pqparse(::Type{Time}, str::AbstractString)` function to use the same approach.

I am using `tryparse` in `catch` block, because for some reason for `ZonedDateTime` `tryparse` is working properly in `1.7`

Arkoniak commented:
Sorry for bothering @iamed2 , but is there anything else that should be done in this PR?
